app/controllers/application_controller.rb

```diff
@@ -1,21 +1,31 @@
 class ApplicationController < ActionController::Base
+  before_action :authenticate_user!
+
   # TODO: Remove when devise is added
-  def user_logged_in?
-    !!current_user
+  def authenticate_user!
+    raise "Not logged in" unless user_logged_in?
   end
 
   def current_user
-    User.first
+    @__current_user__ ||= User.first # rubocop:disable Naming/MemoizedInstanceVariableName
+  end
+
+  def user_logged_in?
+    !!current_user
   end
 
-  %w[index show].each do |action|
-    define_method action do
-      if user_logged_in? && respond_to?("authenticated_#{action}")
-        send("authenticated_#{action}")
-        render action: "authenticated/#{action}"
-      elsif !user_logged_in? && respond_to?("external_#{action}")
-        send("external_#{action}")
-        render action: "external/#{action}"
+  def self.allow_unauthenticated!(*actions)
```

Comment:
Any particular reason why we used this instead of a mixin? I know there really the same under the hood, but just asking for reading purposes :)

Reply:
I figured as it might evolve more etc, I'd wait until we were sure it was stable then we could refactor it.

I did consider using a Concern for it, but figured I'd have to remember how they worked and that sounded like effort :wink:

Once we're happy everything is stable, I'll refactor it out 👍